src/adapters/leasing-adapter.ts

```diff
@@ -562,6 +562,24 @@ const getOffersByListingId = async (
   }
 }
 
+const getActiveOfferByListingId = async (
+  listingId: number
+): Promise<AdapterResult<Offer, 'not-found' | 'unknown'>> => {
+  try {
+    const res = await axios(
+      `${tenantsLeasesServiceUrl}/offers/listing-id/${listingId}/active`
+    )
+
+    if (!res.data.content) {
+      return { ok: false, err: 'not-found' }
```

Comment:
Also same thought on using pre-defined error codes in onecore-types to set the various error code values.

Reply:
Yeah I think adding types for errors for the different endpoints are a good idea but perhaps not needed in the adapters